src/core/query.ts

```diff
@@ -475,7 +475,11 @@ export class Query<
     return {
       data,
       dataUpdateCount: 0,
-      dataUpdatedAt: hasData ? initialDataUpdatedAt || Date.now() : 0,
```

Comment:
Wouldn’t using nullish coalesece solve the same purpose?

```diff
- hasData ? initialDataUpdatedAt || Date.now()
+ hasData ? initialDataUpdatedAt ?? Date.now()
```

Reply:
It would perhaps be the same as the current code on master, but not as the numeric type check.

Reply:
that's true, but what else should it be? The types clearly state that it should be `number | undefined`, and the docs say that it has to be _a JS timestamp in milliseconds_.

I don't think that we need to check every option for every possible input - that's what TypeScript is for. Yes, there are also (lots of) JS consumers, but if I put in a string (`"1608412420052"`) - would it not be weird if that were converted to `Date.now()` ?